torch/distributed/optim/named_optimizer.py

```diff
@@ -191,9 +191,14 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
             )
 
         for idx, param_key in enumerate(self.ordered_param_keys):
-            # When the conditional training is performed, not all parameters are updated in the optim.
+            # All states must exist in the provided ``state_dict``. Even if a
+            # parameter is not used (size is empty or not used in the model),
+            # the initial state should exist in the ``state_dict`` to satisfy
+            # the assumption of NamedOptimizer.
             if param_key not in state.keys():
-                continue
+                raise ValueError(
```

Comment:
What about partial training case which enabled in https://github.com/pytorch/pytorch/pull/91344?